File: 1501-2000/1904.The Number of Full Rounds You Have Played.py
from datetime import datetime

class Solution:
    def numberOfRounds(self, startTime: str, finishTime: str) -> int:
        s = datetime.strptime(startTime, "%H:%M")
        e = datetime.strptime(finishTime, "%H:%M")
        ROUND = 15 * 60
        if 0 <= (e - s).seconds < ROUND:
            return 0
        return int(e.timestamp() // ROUND - (s.timestamp() + ROUND - 60) // ROUND) % (24 * 4)

File: 1501-2000/1904.The Number of Full Rounds You Have Played.ts
function numberOfRounds(startTime: string, finishTime: string): number {
        const getMinute = (time:string):number => parseInt(time.slice(0, 2)) * 60 + parseInt(time.slice(3));
        const ROUND = 15;
        const start = getMinute(startTime);
        const end = getMinute(finishTime);
        const startRound = Math.ceil(start / ROUND);
        const endRound = Math.floor(end / ROUND);
        if (startRound === endRound + 1 && start <= end) {
            return 0;
        }
        return (endRound - startRound + 24 * 4) % (24 * 4);
};

